src/jit/liveness.cpp

```diff
@@ -2359,6 +2359,19 @@ bool Compiler::fgRemoveDeadStore(GenTree**        pTree,
                     printf("\n");
                 }
 #endif // DEBUG
+
+                if (rhsNode->TypeGet() == TYP_STRUCT)
+                {
+                    // This is a block assignment. An indirection of the rhs is not considered to
+                    // happen until the assignment, so we will extract the side effects from only
+                    // the address.
+                    if (rhsNode->OperIsIndir())
+                    {
+                        assert(rhsNode->OperGet() != GT_NULLCHECK);
```

Comment:
Why should not it be a nullcheck?

Reply:
We never create GT_NULLCHECK as a rhs of an assignment. I guess this assert just makes sure we don't accidentally remove a null check.